neo/io/proxyobjects.py

```diff
@@ -588,3 +597,33 @@ def consolidate_time_slice(time_slice, seg_t_start, seg_t_stop, strict_slicing):
     t_stop = ensure_second(t_stop)
 
     return (t_start, t_stop)
+
+
+def create_analogsignal_array_annotations(sig_annotations, global_channel_indexes):
+    """
+    Create array_annotations from raw_annoations.
+    Since raw_annotation are not np.array but nested dict, this func
+    try to find keys in raw_annotation that are shared by all channel
+    and make array_annotation with it.
+    """
+    # intersection of keys across channels
+    common_keys = None
+    for ind in global_channel_indexes:
+        keys = [k for k, v in sig_annotations[ind].items() if not \
+                    isinstance(v, (list, tuple, np.ndarray))]
+        if common_keys is None:
+            common_keys = keys
+        else:
+            common_keys = [k for k in common_keys if k in keys]
+
+    # this is redundant and done at other level
```

Comment:
this comment is either outdated or code needs to be changed here. Which of the two is correct?

Reply:
Or do you want to keep `name`, `channel_id` and `file_origin` here and remove the more specific code at the other location?

Reply:
channel_names and channel_ids are done separatly directly from the header['signal_channel'].
So it is redundant here, this is why they are not included here.

file_origin in array_annoations have no sens. But they are push automatically for all IOs.
It is already in annotations.

Reply:
Maybe I need to reformulate the comments.

Reply:
why does 'file_origin` not make sense? For some IOs all individual traces in an AnalogSignal might come from different files, e.g. NeuralynxIO.

Reply:
because an array per channel of file origin is not util.
file_origin is more a scalar annoataions.
It repeat the same file_origin over channel.
No ?

Reply:
Not in the case of Neuralynx, where you have one file per channel.

Reply:
OK.
I can put it if you want but for the main case it can add a quite heavy array (full X nb_channel) that can be some kb.

Reply:
But it's just an array of strings with length = nb_channels, so it shouldn't be heavy, no?

Reply:
If full path is 150bytes > 1024 channel then it is 150kb.
Which is quite light compare to signal but quqite heavy for a metadata.... :)
As you want.
I not not sure that this metadatat need to be per channel.

Reply:
Can we implement it for sake of completeness now as array annotation and if this turns out to be too heavy in the future we can still revert?

Reply:
OK done